### ISSUE TITLE
IncompatibleClassChangeError with flutter-background-location-tracker

### ISSUE THREAD

**Issue by steffomix** [Baseflow/flutter-geolocator#1233]
The geolocator ^9.0.2 doesn't work together with https://github.com/icapps/flutter-background-location-tracker  ^1.4.0

Resulting in:
> java.lang.IncompatibleClassChangeError: Found interface com.google.android.gms.location.FusedLocationProviderClient, but class was expected (declaration of 'com.google.android.gms.location.FusedLocationProviderClient'

**Using geolocator: 7.4.0 does fix the issue.**

**Build Stacktrace:**

> Launching lib\main.dart on sdk gphone64 x86 64 in debug mode...
> Note: C:\Users\steff\AppData\Local\Pub\Cache\hosted\pub.dev\geolocator_android-4.1.7\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java uses or overrides a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> √  Built build\app\outputs\flutter-apk\app-debug.apk.
> D/FlutterGeolocator(30652): Attaching Geolocator to activity
> D/FlutterGeolocator(30652): Creating service.
> D/FlutterGeolocator(30652): Binding to location service.
> E/AndroidRuntime(30652): FATAL EXCEPTION: main
> E/AndroidRuntime(30652): Process: com.stefanbrinkmann.chaostours, PID: 30652
> E/AndroidRuntime(30652): **java.lang.IncompatibleClassChangeError: Found interface com.google.android.gms.location.FusedLocationProviderClient, but class was expected (declaration of 'com.google.android.gms.location.FusedLocationProviderClient'** appears in /data/app/~~FPuGdmBUE1bN4zoULkYo9Q==/com.stefanbrinkmann.chaostours-Gi1QeKnPNBo1hu-npFKFVQ==/base.apk)
> E/AndroidRuntime(30652): 	at com.icapps.background_location_tracker.service.LocationUpdatesService.getLastLocation(LocationUpdatesService.kt:215)
> E/AndroidRuntime(30652): 	at com.icapps.background_location_tracker.service.LocationUpdatesService.onCreate(LocationUpdatesService.kt:74)
> E/AndroidRuntime(30652): 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4485)
> E/AndroidRuntime(30652): 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
> E/AndroidRuntime(30652): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2161)
> E/AndroidRuntime(30652): 	at android.os.Handler.dispatchMessage(Handler.java:106)
> E/AndroidRuntime(30652): 	at android.os.Looper.loopOnce(Looper.java:201)
> E/AndroidRuntime(30652): 	at android.os.Looper.loop(Looper.java:288)
> E/AndroidRuntime(30652): 	at android.app.ActivityThread.main(ActivityThread.java:7872)
> E/AndroidRuntime(30652): 	at java.lang.reflect.Method.invoke(Native Method)
> E/AndroidRuntime(30652): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
> E/AndroidRuntime(30652): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
> Error connecting to the service protocol: failed to connect to http://127.0.0.1:50614/FSC88d7HPmI=/
> Exited (sigterm)

### Expected behavior
None. crashes during compile process.

### Reproduction steps
Add packages to your Android project:
- https://pub.dev/packages/background_location_tracker ^1.4.0
- https://pub.dev/packages/geolocator ^9.0.2
- Try to compile debug for android with flutter run

### Configuration
sdk: '>=2.19.4 <3.0.0'

**Platform:** 
- [x] :robot: Android 11, API Level 33

**Comment by Samuel-ML** [comment]
same errror

**Comment by steffomix** [comment]
I found two ways to solve the issue with a workaround.
The first way is to downgrade geolocator. The other is by downloading the latest version of background location tracker directly from github and import it localy.
So that the result may look like that:
```
dependencies:
  # https://github.com/icapps/flutter-background-location-tracker
  background_location_tracker:
    path: ../flutter-background-location-tracker

  #geolocator: 7.4.0 # version 9.x collides with background_location_tracker
  geolocator: 9.0.2
```

**Comment by mvanbeusekom** [comment]
This issue was caused by the `background_location_tracker` plugin using an outdated version of Androids `com.google.android.gms:play-services-location` library. This has been fixed in version 1.4.1 of the `background_location_tracker`.
